### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Nuru**
## what

- Module for creating AWS IAM roles for Kubernetes service accounts under EKS

## why

- Give service accounts access to resources they need without giving them or other actors (such as EC2 instances) the same access

### CODE REVIEW

File: .github/workflows/chatops.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  issue_comment:
+    types: [created]
+
+jobs:
+  default:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Handle common commands"
+        uses: cloudposse/actions/github/slash-command-dispatch@0.15.0
```
Comment (aknysh) [inline .github/workflows/chatops.yml]:
```suggestion
        uses: cloudposse/actions/github/slash-command-dispatch@0.16.0
```

File: .github/workflows/chatops.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          permission: none
+          issue-type: pull-request
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout commit"
+        uses: actions/checkout@v2
+      - name: "Run tests"
+        uses: cloudposse/actions/github/slash-command-dispatch@0.15.0
```
Comment (aknysh) [inline .github/workflows/chatops.yml]:
```suggestion
        uses: cloudposse/actions/github/slash-command-dispatch@0.16.0
```

File: README.yaml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    url: "https://cloudposse.com"
+    year: "2020"
+
+# Canonical GitHub repo
+github_repo: cloudposse/terraform-aws-eks-iam-role
+
+# Badges to display
+badges:
+  - name: "Latest Release"
+    image: "https://img.shields.io/github/release/cloudposse/terraform-example-module.svg"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    image: "https://img.shields.io/github/release/cloudposse/terraform-aws-eks-iam-role.svg"
```

File: README.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    url: "https://github.com/cloudposse/terraform-example-module/releases/latest"
```
Comment (aknysh) [inline README.yaml]:
```suggestion
    url: "https://github.com/cloudposse/tterraform-aws-eks-iam-role/releases/latest"
```

File: README.yaml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    namespace   = var.namespace
+    environment = var.environment
+    stage       = var.stage
+    name        = var.name
+    delimiter   = var.delimiter
+    attributes  = var.attributes
+    tags        = var.tags
+
+    aws_account_number          = local.account_id
+    eks_cluster_oidc_issuer_url = module.eks_cluster.eks_cluster_identity_oidc_issuer
```
Comment (aknysh) [inline README.yaml:84]: do we need to remove `https://` from the issue URL (I don't remember)

Comment (Nuru) [reply README.yaml:84]: Yes, but the module does that

File: README.yaml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    url: "https://ask.sweetops.com/"
+
+# List any related terraform modules that this module may be used with or that this module depends on.
+related:
+  - name: "terraform-null-label"
+    description: "Terraform module designed to generate consistent names and tags for resources. Use terraform-null-label to implement a strict naming convention."
+    url: "https://github.com/cloudposse/terraform-null-label"
+
+# List any resources helpful for someone to get started. For example, link to the hashicorp documentation or AWS documentation.
+references:
```
Comment (osterman) [inline README.yaml:44]: In a follow-up PR, please manually update this list to point to related terraform resources that are used or anything else that will help the user understand the module

File: README.yaml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  - name: "Slack Community"
+    image: "https://slack.cloudposse.com/badge.svg"
+    url: "https://slack.cloudposse.com"
+  - name: "Discourse Forum"
+    image: "https://img.shields.io/discourse/https/ask.sweetops.com/posts.svg"
+    url: "https://ask.sweetops.com/"
+
+# List any related terraform modules that this module may be used with or that this module depends on.
+related:
+  - name: "terraform-null-label"
```
Comment (osterman) [inline README.yaml:39]: In a follow-up PR, please manually update this list to point to related modules that provide IAM and EKS capabilities.

**Review by osterman — APPROVED** [review]

**Review by aknysh — APPROVED** [review]

**Pull request merged**